desktop/core/src/desktop/js/components/AlertBox.vue

```diff
@@ -0,0 +1,191 @@
+  distributed with this work for additional information
+  regarding copyright ownership.  Cloudera, Inc. licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<template>
+  <div :class="['alert-box', definition.cssClass]">
+    <em v-if="showClose" class="alert-box-close fa fa-times alert-color" @click="$emit('close')" />
+
+    <em :class="['fa alert-icon alert-color', definition.faClass]" />
+    <div class="alert-box-title">{{ definition.title }}</div>
+
+    <span class="alert-box-message">
+      {{ message }}
+    </span>
+
+    <div v-if="showDetails">
+      <a v-if="details" class="more-less-button" @click="setShowDetails(false)">
+        Less details
+      </a>
+      <pre class="alert-box-details">
+        {{ details }}
+      </pre>
+    </div>
+    <a v-else-if="details" class="more-less-button" @click="setShowDetails(true)">
+      More details
+    </a>
+  </div>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import Component from 'vue-class-component';
+  import { Prop } from 'vue-property-decorator';
+
+  export enum AlertType {
+    Success,
+    Error,
+    Warning,
+    Info,
+    Unknown
+  }
+
+  class TypeDefinition {
+    title: string;
+    faClass: string;
+    cssClass: string;
+
+    constructor(title: string, faClass: string, cssClass: string) {
+      this.title = title;
+      this.faClass = faClass;
+      this.cssClass = cssClass;
+    }
+  }
+
+  const UNKNOWN_DEF = new TypeDefinition('Unknown!', 'fa-question-circle', 'ea-unknown');
+  const TYPE_DEFS: Map<AlertType, TypeDefinition> = new Map([
+    [AlertType.Success, new TypeDefinition('Success!', 'fa-check-circle', 'ea-success')],
+    [AlertType.Error, new TypeDefinition('Error', 'fa-exclamation-circle', 'ea-error')],
+    [AlertType.Warning, new TypeDefinition('Warning', 'fa-exclamation-triangle', 'ea-warning')],
+    [AlertType.Info, new TypeDefinition('Info', 'fa-info-circle', 'ea-info')],
+    [AlertType.Unknown, UNKNOWN_DEF]
+  ]);
+
+  @Component
+  export default class AlertBox extends Vue {
+    @Prop({ required: true, default: AlertType.Unknown })
+    type!: AlertType;
+
+    @Prop()
+    message!: string;
+    @Prop()
+    details!: string;
+
+    @Prop({ default: false })
+    showClose!: boolean;
+
+    showDetails = false;
+
+    get definition(): TypeDefinition {
+      return TYPE_DEFS.get(this.type) || UNKNOWN_DEF;
+    }
+
+    setShowDetails(show: boolean): void {
```

Comment:
Nit: no need for the function as can do showDetails = x above, or change to toggleDetails(): void

Reply:
showDetails = x looks sweet 😍